src/lidar_tools/dsm_functions.py

```diff
@@ -50,51 +85,122 @@ def return_readers(input_aoi,
+                # now convert the buffered tile to a polygon
+                aoi_3857 = shapely.geometry.Polygon.from_bounds(*aoi_3857_bounds)
+                print(f"The buffered tile bound is: ", aoi_3857.bounds)
+                
 
 
             gdf = gpd.read_file('https://raw.githubusercontent.com/hobuinc/usgs-lidar/master/boundaries/resources.geojson').set_crs(4326)
             # in the eventuality that the above URL breaks, we store a local copy
             # gdf = gpd.read_file('../data/shapefiles/resources.geojson').set_crs(4326)
-
+            if return_specific_3dep_survey is not None:
+                return_all_intersecting_surveys = True
             for _, row in gdf.iterrows():
                 if row.geometry.intersects(aoi_4326):
                     usgs_dataset_name = row['name']
-                    break
-
-            print("Dataset being used: ", usgs_dataset_name)
-            url = f"https://s3-us-west-2.amazonaws.com/usgs-lidar-public/{usgs_dataset_name}/ept.json"
-            reader = {
-            "type": "readers.ept",
-            "filename": url,
-            "resolution": pointcloud_resolution,
-            "polygon": str(aoi_3857.wkt),
-            }
-
-            # SRS associated with the 3DEP dataset
-            response = requests.get(url)
-            data = response.json()
-            srs_wkt = data['srs']['wkt']
-
-            pointcloud_input_crs.append(CRS.from_wkt(srs_wkt))
-            readers.append(reader)
+                    if return_specific_3dep_survey is not None:
+                        if usgs_dataset_name == return_specific_3dep_survey:
+                            add_survey = True
+                        else:
+                            add_survey = False
+                    else:
+                        add_survey = True
+                    if add_survey:
+                        print("Dataset being used: ", usgs_dataset_name)
+                        url = f"https://s3-us-west-2.amazonaws.com/usgs-lidar-public/{usgs_dataset_name}/ept.json"
+                        reader = {
+                        "type": "readers.ept",
+                        "filename": url,
+                        "resolution": pointcloud_resolution,
+                        "polygon": str(aoi_3857.wkt),
+                        }
+
+                        # SRS associated with the 3DEP dataset
+                        response = requests.get(url)
+                        data = response.json()
+                        srs_wkt = data['srs']['wkt']
+
+                        pointcloud_input_crs.append(CRS.from_wkt(srs_wkt))
+                        readers.append(reader)
+                        extents.append(aoi_3857.bounds)
+                        original_extents.append(src_bounds_transformed_3857)
+                    if not return_all_intersecting_surveys:
+                        break
+
+            
+
+    return readers, pointcloud_input_crs, extents, original_extents
```

Comment:
CI is failing (https://github.com/uw-cryo/lidar_tools/actions/runs/15000341829/job/42145345525?pr=9) because this function now returns more things. Would be good to update the docstring for this function to clarify it now also returns extents and original extents. Feel free to change the test here to check for these new outputs:

https://github.com/uw-cryo/lidar_tools/blob/6b5d1fba242d4649669cc96b76b8217539b6a97a/tests/test_all.py#L14

Reply:
Docstring updated in: [1f9ca8c](https://github.com/uw-cryo/lidar_tools/commit/1f9ca8ca25240aeb4edecbd67822e163ade10700)

Tried to update test here: [41fb7d5](https://github.com/uw-cryo/lidar_tools/commit/41fb7d51e9e8944a5db3c2d58a273bb441b6ca40)